deepspeed/runtime/zero/parameter_offload.py

```diff
@@ -392,7 +392,8 @@ def _run_before_forward_function(input):
                                                                _run_after_backward_hook, inputs)
 
         def _post_backward_module_hook(module, inputs):
-            module.ds_grads_remaining = 0
+            if not hasattr(module, "ds_grads_remaining"):
```

Comment:
Can you please explain why `hasattr` is needed here?

Reply:
Also copy to here:
"ds_grads_remaining" is used to triger post_backward_function(). If the
module is called more than once in one training step, this variable will
be initialized every time. If backward() is also called multiple times,
"ds_grads_remaining" will be reduced to a negative number.
post_backward_function() will not be called as expected. This leads to
extra fetch operation or extra memory usage.

Reply:
Use "hasattr" to avoid repeat initialization.